pkg/util/aad/aad_test.go

```diff
@@ -0,0 +1,134 @@
+package aad
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_refreshable "github.com/Azure/ARO-RP/pkg/util/mocks/refreshable"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+func Test_refreshContext(t *testing.T) {
+	mockController := gomock.NewController(t)
+	var done bool
+	type args struct {
+		authorizer refreshable.Authorizer
+		mocksetup  func(refreshable.Authorizer)
+		ctx        context.Context
+		done       *bool
+		log        *logrus.Entry
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr error
+	}{
+		{
+			name: "RefreshWithContext without errors",
+			args: args{
+				authorizer: mock_refreshable.NewMockAuthorizer(mockController),
+				mocksetup: func(refreshauth refreshable.Authorizer) {
+					refreshauth.(*mock_refreshable.MockAuthorizer).EXPECT().RefreshWithContext(gomock.Any(), gomock.Any()).Return(true, nil).AnyTimes()
+				},
+				ctx:  context.Background(),
+				done: &done,
+			},
+			wantErr: nil,
+		},
+		{
+			name: "RefreshWithContext with azure errors",
+			args: args{
+				authorizer: mock_refreshable.NewMockAuthorizer(mockController),
+				mocksetup: func(refreshauth refreshable.Authorizer) {
+					refreshauth.(*mock_refreshable.MockAuthorizer).EXPECT().RefreshWithContext(gomock.Any(), gomock.Any()).Return(false, nil).AnyTimes()
+				},
+				ctx:  context.Background(),
+				done: &done,
+			},
+			wantErr: nil,
+		},
+		{
+			name: "RefreshWithContext with errors",
+			args: args{
+				authorizer: mock_refreshable.NewMockAuthorizer(mockController),
+				mocksetup: func(refreshauth refreshable.Authorizer) {
+					refreshauth.(*mock_refreshable.MockAuthorizer).EXPECT().RefreshWithContext(gomock.Any(), gomock.Any()).Return(false, errors.New("Azure error")).AnyTimes()
+				},
```

Comment:
@facchettos : Seems like this is the pattern that we follow, could you please refer below test cases:

- pkg/util/instancemetadata/prod_test.go
- pkg/util/steps/runner_test.go

Reply:
I know we are using that pattern some other places. But it isn't a great pattern, so since this is a new file and we are creating the tests, it is better not to use it. Table driven tests are great because we can feed parameters and see in one glance what the test parameters are, this pattern is actually breaking that and this is why we should not use it for new test. 

Here it is bad to use this because it allows for subtest that are widely different from each other, and that should be in different tests. Also, this is duplicating code and now if we are to add new parameters or anything, we need to change all of the functions, while we wouldn't have to do it if that part was directly in the test and we were feeding the parameters to it. Code should only contain the logic, structs are for data. 

Also even though I think you should remove the functions altogether, I am not sure why your functions are using `refreshable.Authorizer` but you only give them `*mock_refreshable.MockAuthorizer` as parameters, and then always typecast them back to the latter.